File: smk_scm/public/py/payment_entry.py

import frappe
from frappe.email.queue import flush
from frappe.utils import get_fullname

@frappe.whitelist()
def send_advance_payment_email(purchase_order, recipients):
    recipients = frappe.parse_json(recipients)
    try:
        # Construct email content
        subject = f"Advance Payment Notification for Purchase Order {purchase_order}"
        message = f"Dear Team,\n\nPlease be informed that an advance payment has been made for Purchase Order {purchase_order}.\nPlease take the necessary actions.\n\nThank you."

        # Send email to recipients
        frappe.sendmail(
            recipients=recipients,
            subject=subject,
            message=message
        )
        flush()

        return "Email sent successfully"
    
    except Exception as e:
        frappe.log_error(f"Failed to send email for Purchase Order {purchase_order}: {str(e)}")
        return "Failed to send email"

File: smk_scm/public/js/payment_entry.js
frappe.ui.form.on('Payment Entry', {
    refresh: function(frm) {
        frm.doc.references.forEach(reference => {
            if (reference.reference_doctype === "Purchase Order" && reference.reference_name) {
                frappe.db.get_value("Purchase Order", reference.reference_name, ["payment_terms_template"], (r) => {
                    if (r && r.payment_terms_template === "Advance Payment") {
                        const recipients = ["purchase_team@gmail.com", "logistic_team@gmail.com"];

                        if (frm.doc.contact_email) {
                            recipients.push(frm.doc.contact_email);
                        }

                        frappe.call({
                            method: 'smk_scm.public.py.payment_entry.send_advance_payment_email',
                            args: {
                                purchase_order: reference.reference_name,
                                recipients: recipients
                            },
                            callback: function(response) {
                                if (response.message === 'Email sent successfully') {
                                    frappe.show_alert({
                                        message: 'Advance payment notification email sent.',
                                        indicator: 'green'
                                    });
                                } else {
                                    frappe.show_alert({
                                        message: 'Failed to send email.',
                                        indicator: 'red'
                                    });
                                }
                            }
                        });
                    }
                });
            }
        });
    }
});
